File: ios/AppInfo/AppInfo.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE (AppInfo, NSObject)

RCT_EXTERN_METHOD(getAppInfo : (RCTPromiseResolveBlock)
                      resolve rejecter : (RCTPromiseRejectBlock)reject)

@end

File: ios/AppInfo/AppInfo.swift
@objc(AppInfo)
class AppInfo: NSObject {

  @objc
  func getAppInfo(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {
    let info = Bundle.main.infoDictionary

    resolve([
      "version": info?["CFBundleShortVersionString"] as? String ?? "",
      "buildNumber": info?["CFBundleVersion"] as? String ?? "",
      "platform": "ios"
    ])
  }
}
